### PR TITLE
Make the remaining management buttons work like links

### PR DESCRIPTION
**Pull request by FrankeLukas**
### Checklist

- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).

### Motivation and Context

Continues #2669. All other "buttons" are not router links and would need even further refactoring.

### Steps for Testing

The following buttons did change and work like links now:
- Learning Goals: "Edit", "Create"
- (Test) Exam summary: "View Results"
- Exam management: "Exam checklist"
- Text & File Upload Instructor Assessment (Submissions) Dashboard (Go to "Instructor Dashboard" then to "Submissions"): "Open Assessment"
- Programming exercises: "Edit in Editor", "Check Plagiarism", "Generate a new Programming Exercise"
- Exercise Hints: "Edit"
- Exercise Scores: "Participations", "View Assessment"

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/exercise-hint/manage/exercise-hint-detail.component.html [hunk 1]
```diff
@@ -21,9 +21,9 @@ <h2><span jhiTranslate="artemisApp.exerciseHint.detail.title">Exercise Hint</spa
                 </dd>
             </dl>
 
-            <button type="button" [routerLink]="['edit']" class="btn btn-primary">
+            <a type="button" [routerLink]="['edit']" class="btn btn-primary">
```
Comment (iozbeyli) [inline src/main/webapp/app/exercises/shared/exercise-hint/manage/exercise-hint-detail.component.html:24]: Anchor tag type button is not necessary.

File: src/main/webapp/app/exercises/text/assess/text-submission-assessment.component.html [hunk 2]
```diff
@@ -62,9 +62,9 @@
 <ng-template #noSubmission>
     <div *ngIf="noNewSubmissions" class="alert alert-warning text-center mt-4" role="alert">
         <p jhiTranslate="artemisApp.textAssessment.notFound">We haven't found any new text submission without an assessment, please go back.</p>
-        <button type="submit" [routerLink]="['../../../../../exercises', exerciseId, 'tutor-dashboard']" class="btn btn-info btn-sm mr-1 mb-1 tutor-dashboard">
+        <a type="submit" [routerLink]="['../../../../../exercises', exerciseId, 'tutor-dashboard']" class="btn btn-info btn-sm mr-1 mb-1 tutor-dashboard">
```
Comment (iozbeyli) [inline src/main/webapp/app/exercises/text/assess/text-submission-assessment.component.html:65]: Type attribute for anchor means MIME type.

**Comment by FrankeLukas** [comment]
Removed the MIME types I missed.

**Review by airelawaleria — COMMENTED** [review]
Programming exefcise -> Scores -> Participations just opens a new empty window

**Review by JohannesWeiss — COMMENTED** [review]
Tested in testing session, can confirm the issue with the buttons for opening an assessment. Does not work for Text/ Upload exercises as well.

**Review by simonzilker — COMMENTED** [review]
Can confirm the mentioned issues

**Review by airelawaleria — DISMISSED** [review]
The previously mentioned issue wasn't explicitly mentioned in the steps, so approve (may not be yet implemented)

**Review by marvinchimeka — DISMISSED** [review]
Can confirm mentioned issue. But was not stated in the testing steps, so approve

**Review by simonzilker — DISMISSED** [review]
what @airelawaleria said :)

**Review by JohannesWeiss — APPROVED** [review]

**Review by valeriebucher — APPROVED** [review]
Apart from the above mentioned issue, it looks good

**Review by tiffca — DISMISSED** [review]
all buttons worked as expected

**Review by EneaGore — APPROVED** [review]

**Review by nicolasruscher — APPROVED** [review]
Code looks good to me :)

**Comment by FrankeLukas** [comment]
Thanks for the reports everyone. There were indeed bugs hiding there which I fixed in this PR too, now.

**Review by philgabr — APPROVED** [review]
Looks good :+1:

Maybe it would be reasonable to add the "link functionality" to "open assessment" in prog. exercises and to "start new assessment" in general, as well

**Comment by FrankeLukas** [comment]
Unfortunately I couldn't change those since they run client side logic *before* switching the page. Opening a new tab and then running that logic depending on user input doesn't look feasible/possible to me.

**Review by sebastiantpeter — APPROVED** [review]
Tested successfully on ts2.
I noticed that some buttons work like links in Artemis, which should actually just be buttons. For example the "Edit" and "Preview" buttons for programming exercise problem statements. The provide a link to the starting page, what does not really make sense. However, this is not caused by this PR.

**Comment by krusche** [comment]
> Tested successfully on ts2.
> I noticed that some buttons work like links in Artemis, which should actually just be buttons. For example the "Edit" and "Preview" buttons for programming exercise problem statements. They provide a link to the starting page, which does not really make sense. However, this is not caused by this PR.
> buttons_links

We should fix this

**Comment by FrankeLukas** [comment]
Resolved the merge conflicts.

Apparently those edit and preview buttons were "broken" in that regard ever since #1381 changed the markdown editor to use ngbNav. This doesn't look easily fixable and means rewriting a good part of the markdown editor, so I'd like to leave that to a future PR.

**Review by philgabr — APPROVED** [review]
Approve new build :+1:

**Review by airelawaleria — APPROVED** [review]

**Review by EneaGore — APPROVED** [review]
Tested succesfully on ts1. 
Small thing: the username on the top right corner also has a "void(0);" link and opens an empty window about:blank#blocked

**Review by tiffca — APPROVED** [review]
Works as described

**Review by DenizShabani — APPROVED** [review]

**Review by nicolasruscher — APPROVED** [review]
Nice changes! Looks good to me. But maybe we could add a few client tests to check additionally check the routing?

**Pull request merged**